packages/medusa/src/services/payment-collection.ts

```diff
@@ -337,7 +337,7 @@ export default class PaymentCollectionService extends TransactionBaseService {
           const paymentProviderTx =
             this.paymentProviderService_.withTransaction(manager)
 
-          promiseAll(
+          await promiseAll(
```

Comment:
comment: Sorry, I was too quick with my review. Don't think it's needed. I believe we are `.then` chaining here to catch the error and proceed.

Reply:
let me re check

Reply:
No I think you were right, the await was missing

Reply:
since we are in a transaction, i would still wait to finish to prevent the transaction to end before it finishes. wdyt?

Reply:
Yep, I think that sounds reasonable